Tallies/TallyMaps/Maps1D/directionMap_class.f90

```diff
@@ -0,0 +1,194 @@
+    character(nameLen)                       :: type
+    character(100), parameter :: Here = 'init (directionMap_class.f90)'
+
+    ! Check orientation of the cylinder
+    if (dict % isPresent('plane')) then
+      call dict % get(type, 'plane')
+    else
+      type = 'xy'
+    end if
+
+    select case(type)
+      case('yz')
+        self % DIM1 = Y_AXIS
+        self % DIM2 = Z_AXIS
+
+      case('xz')
+        self % DIM1 = X_AXIS
+        self % DIM2 = Z_AXIS
+
+      case('xy')
+        self % DIM1 = X_AXIS
+        self % DIM2 = Y_AXIS
+
+      case default
+        call fatalError(Here, 'Keyword orientation must be x, y or z. It is: '//type)
+
+    end select
+
+    ! Build grid
+    call dict % get(self % N, 'N')
+    call self % bounds % init(-PI, PI, self % N, 'lin')
+
+  end subroutine init
+
+  !!
+  !! Return total number of bins in this division along Dimension D
+  !!
+  !! See tallyMap for specification.
+  !!
+  elemental function bins(self, D) result(N)
+    class(directionMap), intent(in) :: self
+    integer(shortInt), intent(in)   :: D
+    integer(shortInt)               :: N
+
+    if (D == 1 .or. D == 0) then
+      N = self % N
+    else
+      N = 0
+    end if
+
+  end function bins
+
+  !!
+  !! Return string that describes variable used to divide event space
+  !!
+  !! See tallyMap for specification
+  !!
+  function getAxisName(self) result(name)
+    class(directionMap), intent(in)  :: self
+    character(nameLen)               :: name
+
+    name = 'directionMap'
+
+  end function getAxisName
+
+  !!
+  !! Map particle to a single bin. Return 0 for particle out of division
+  !!
+  !! See tallyMap for specification.
+  !!
+  elemental function map(self, state) result(idx)
+    class(directionMap), intent(in)  :: self
+    class(particleState), intent(in) :: state
+    integer(shortInt)                :: idx
+    real(defReal)                    :: x, y, theta
+
+    x = state % dir(self % DIM1)
+    y = state % dir(self % DIM2)
+
+    theta = atan2(y,x)
+    ! Search along the azimuthal dimension and return 0 if index is out-of-bounds
+    idx = self % bounds % search(theta)
+
+    ! Should never happen
+    if (idx == valueOutsideArray) then
+      idx = 0
+      return
+    end if
+
+  end function map
+
+  !!
+  !! Add information about division axis to the output file
+  !!
+  !! See tallyMap for specification.
+  !!
+  subroutine print(self,out)
+    class(directionMap), intent(in)  :: self
+    class(outputFile), intent(inout) :: out
+    character(nameLen)               :: name
+    integer(shortInt)                :: i
+
+    ! Print grid bins
+    name = trim(self % getAxisName()) //'AngularBounds'
+
+    call out % startArray(name, [2,self % N])
+
+    do i = 1, self % N
+      ! Print lower bin boundary
+      call out % addValue(self % bounds % bin(i) + PI)
+
+      ! Print upper bin boundar
+      call out % addValue(self % bounds % bin(i + 1) + PI)
+    end do
```

Comment:
```suggestion
    do i = 1, self % N
      ! Print lower bin boundary
      call out % addValue(self % bounds % bin(i))

      ! Print upper bin boundar
      call out % addValue(self % bounds % bin(i + 1))
    end do
```
NOTE: otherwise the bins don't coincide with the bins of the results.